### ISSUE TITLE
Issue with the convert task when we upload a file

### ISSUE THREAD

**Issue by olivier34000** [cloudconvert/cloudconvert-php#78]
In https://github.com/cloudconvert/cloudconvert-php/blob/986f2c2666c96175d228d6d69561b67749638b76/src/Transport/HttpTransport.php#L185-L191 when a file is uploaded the options is not set in addResource. In the case where we work with a resource the filename is not set. In your ui the result is

```json
{
    "files": [
        {
            "filename": "${filename}"
        }
    ]
}
```

The workaround to fix the issue was to wrap the resource with `new \GuzzleHttp\Psr7\Stream\($resource, ['metadata' => ['uri' => $filename]]);` 

Possible to fix the lib or to fix your service because the conversion does not work although the upload is OK ?

**Comment by josiasmontag** [comment]
How do you create the stream? How does it differ from https://github.com/cloudconvert/cloudconvert-php/blob/986f2c2666c96175d228d6d69561b67749638b76/tests/Integration/TaskTest.php#L36?

**Comment by olivier34000** [comment]
The steam is retrieved from Openstack. In the test `stream_get_meta_data` (from MultipartStreamBuilder) retrieve the pathname 
```php

$fp = fopen(__DIR__ . '/files/input.pdf', 'r');
print_r(stream_get_meta_data($fp));

Array
(
    [timed_out] => 
    [blocked] => 1
    [eof] => 
    [wrapper_type] => plainfile
    [stream_type] => STDIO
    [mode] => r
    [unread_bytes] => 0
    [seekable] => 1
    [uri] => /tmp/cloudconvert/vendor/cloudconvert/cloudconvert-php/tests/Integration/files/input.pdf
```
However in my case the uri is not set. It comes from php://temp. That's why I have the issue. The metadata from the file
```
Array
(
    [wrapper_type] => PHP
    [stream_type] => TEMP
    [mode] => w+b
    [unread_bytes] => 0
    [seekable] => 1
    [uri] => php://temp
)
```

**Comment by le-yak** [comment]
I believe #82 fixes this issue (ping @josiasmontag), `TasksResource::upload()` will accept a third optional `string $fileName` parameter.

**Comment by josiasmontag** [comment]
Yes, should be fixed with version 3.2.1.
